src/main/java/org/myrobotlab/service/IntegratedMovement.java

```diff
@@ -511,7 +511,7 @@ public Thread getEngine(String arm) {
   }
 
   @Override
-  public Map<String, Double> publishJointAngles(HashMap<String, Double> angleMap) {
+  public AngleData publishJointAngles(AngleData angleMap) {

```

Comment:
If we only publish a single value in the AngleData this method should be publishJointAngle .. not plural. Also, AngleData is not a "map"

Reply:
fixed